clients/notification-kafka/src/main/java/org/eclipse/hono/client/notification/KafkaBasedNotificationSender.java

```diff
@@ -0,0 +1,123 @@
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.client.notification;
+
+import java.net.HttpURLConnection;
+import java.util.Objects;
+
+import org.eclipse.hono.client.ServerErrorException;
+import org.eclipse.hono.client.kafka.producer.KafkaProducerConfigProperties;
+import org.eclipse.hono.client.kafka.producer.KafkaProducerFactory;
+import org.eclipse.hono.notification.AbstractNotification;
+import org.eclipse.hono.notification.NotificationSender;
+import org.eclipse.hono.notification.deviceregistry.CredentialsChangeNotification;
+import org.eclipse.hono.notification.deviceregistry.DeviceChangeNotification;
+import org.eclipse.hono.notification.deviceregistry.TenantChangeNotification;
+
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.json.JsonObject;
+import io.vertx.kafka.client.producer.KafkaProducerRecord;
+import io.vertx.kafka.client.producer.RecordMetadata;
+
+/**
+ * A client for publishing notifications with Kafka.
+ */
+public class KafkaBasedNotificationSender implements NotificationSender {
+
+    public static final String PRODUCER_NAME = "notification";
+    private final KafkaProducerConfigProperties config;
+    private final KafkaProducerFactory<String, JsonObject> producerFactory;
+    private boolean stopped = false;
+
+    /**
+     * Creates an instance.
+     *
+     * @param producerFactory The factory to use for creating Kafka producers.
+     * @param kafkaProducerConfig The Kafka producer configuration properties to use.
+     * @throws NullPointerException if any of the parameters are {@code null}.
+     */
+    public KafkaBasedNotificationSender(final KafkaProducerFactory<String, JsonObject> producerFactory,
+            final KafkaProducerConfigProperties kafkaProducerConfig) {
+        Objects.requireNonNull(producerFactory);
+        Objects.requireNonNull(kafkaProducerConfig);
+
+        this.producerFactory = producerFactory;
+        this.config = kafkaProducerConfig;
+    }
+
+    @Override
+    public Future<Void> publish(final AbstractNotification notification) {
+
+        Objects.requireNonNull(notification);
+
+        if (stopped) {
+            return Future.failedFuture(
+                    new ServerErrorException(HttpURLConnection.HTTP_UNAVAILABLE, "sender already stopped"));
+        }
+
+        try {
+
+            final String topic = NotificationTopicHelper.getTopicName(notification.getClass());
+            final String key = getKey(notification);
+            final JsonObject value = JsonObject.mapFrom(notification);
+            final KafkaProducerRecord<String, JsonObject> record = KafkaProducerRecord.create(topic, key, value);
+
+            final Promise<RecordMetadata> sendPromise = Promise.promise();
+            producerFactory.getOrCreateProducer(PRODUCER_NAME, config).send(record, sendPromise);
```

Comment:
will this create the topic implicitly or does the topic need to be created upfront?

Reply:
This depends on the broker configuration. I would add that to the documentation.

Reply:
The documentation will be in another PR, I guess?

Reply:
Yes. I think this is a concern of the component that publishes a notification.